### ISSUE TITLE
Page promotions ordering

### ISSUE THREAD

**Issue by alxpolyakov** [django-oscar/django-oscar#2267]
promotions.views.PageDetailView  has a bug.
```python
    def post(self, request, **kwargs):
        """
        When called with a post request, try and get 'promo[]' from
        the post data and use it to reorder the page content blocks.
        """
        data = dict(request.POST).get('promo[]')
        self._save_page_order(data)
        return HttpResponse(status=200)
```
causes an Exception. POST is a QueryDict({'promo': [1, 2, 3...] })

this works:
```python
    def post(self, request, **kwargs):
        """
        When called with a post request, try and get 'promo[]' from
        the post data and use it to reorder the page content blocks.
        """
        data = request.POST.getlist('promo')
        data = {data[x]: x for x in range(len(data))}

        self._save_page_order(data)
        return HttpResponse(status=200)
```

**Comment by sasha0** [comment]
Thanks for the report. PR is welcomed.

**Comment by alxpolyakov** [comment]
this does not works too.
this works
`def post(self, request, **kwargs):
        """
        When called with a post request, try and get 'promo[]' from
        the post data and use it to reorder the page content blocks.
        """
        data = request.POST.getlist('promo')
        self._save_page_order(data)
        return HttpResponse(status=200)`

**Comment by SalahAdDin** [comment]
@alxpolyakov make a pr please.

**Comment by solarissmoke** [comment]
I cannot reproduce this - the existing code works fine for me. Please can you post a full stack trace of your error with steps to reproduce?

**Comment by sasha0** [comment]
Already fixed by PR #2247.
